utils/githooks/pre-commit.d/70-isort.sh

```diff
@@ -1,25 +1,36 @@
 #!/bin/bash
-
+#
+# Copyright 2023-2024 Intel Corporation.
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
 # Runs isort for the DAOS project.
 
 set -ue
 
+_print_githook_header "isort"
+
+py_files=$(_git_diff_cached_files "*.py SConstruct *SConscript")
+
+if [ -z "$py_files" ]; then
+    echo "No python changes. Skipping"
+    exit 0
+fi
+
 if ! command -v isort > /dev/null 2>&1; then
-    echo "No isort checking, install isort command to improve pre-commit checks"
+    echo "isort not installed. Install isort command to improve pre-commit checks:"
+    echo "  python3 -m pip install -r ./utils/cq/requirements.txt"
     . /etc/os-release
     if [ "$ID" = "fedora" ]; then
-        echo "dnf install python3-isort"
-        echo "or"
+        echo "  or"
+        echo "  dnf install python3-isort"
     fi
-    echo "python3 -m pip install -r ./utils/cq/requirements.txt"
     exit 0
 fi
 
-echo "isort:"
+echo "Checking if python imports are sorted"
 
-echo "  Running isort for python imports."
-isort_args=(--jobs 8 .)
-if ! isort --check-only "${isort_args[@]}"; then
-    echo "  isort check failed, run 'isort ${isort_args[*]}' to fix."
+if ! echo "$py_files" | xargs -r isort --check-only --jobs 8; then
+    echo "  isort check failed, run 'isort --jobs 8 .' to fix."
```

Comment:
isort was previously running against the entire source tree even if no python changes were made. On my boro node, this took ~1.5s